R/boxr_s3_classes.R

```diff
@@ -303,6 +315,30 @@ summary.boxr_dir_comparison <- function(object, ...) {
   invisible(object)
 }
 
+# Collab Functions -----------------------------------------------
+
+#' @importFrom tibble as_tibble
+#' @export
+as_tibble.boxr_collab <- function(x, ...) {
+  stack_row(x)
+}
+
+#' @export
+as.data.frame.boxr_collab <- function(x, ...) {
+  as.data.frame(as_tibble(x))
+}
+
+# Collab-list Functions -----------------------------------------------
+
+#' @export
+as_tibble.boxr_collab_list <- function(x, ...) {
+  stack_rows(x$entries)
+}
+
+#' @export
+as.data.frame.boxr_collab_list <- function(x, ...) {
+  as.data.frame(as_tibble(x))
```

Comment:
This forces us to become dependent of `tibble`, which I'm not opposed, just want to pause for consideration. We already use `dplyr` and `purrr` and obviously the `tidyverse` is great, but I am leery of depending on it production/stable code. We just got burned by `bind_rows()` switching to new `vctrs` coercion rules that don't sync with base R rules. I love the tidyverse, but want to make sure we acknowledge using more and more of it, will lead to more and more "maintanence" chores for us.

Reply:
I'm torn on this - on the one hand, I'd like to remove the dplyr dependency, i.e. make it a suggests - on the other hand, I think the tidyverse code we write is cleaner and more understandable.

At the risk of drawing out this PR (which I think has become a very useful exercise for us both), maybe this is the time to write that helper-function in base R, then use *that* as the `as.data.frame()` method. 

If so, in the interest of consistency with what Brendan has already done - perhaps this stacking function could follow his convention for the "nested" variable names.

A base function would allow us, eventually, to offer dplyr as a "suggests", and `as_tibble()` for those who want it. Maybe `box_wrangle()` could return an opinionated form of the data.